pkg/server/plugin/bundlepublisher/bundleplublisher.go

```diff
@@ -0,0 +1,14 @@
+package bundlepublisher
+
+import (
+	"context"
+
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+type BundlePublisher interface {
+	catalog.PluginInfo
+
+	PublishBundle(ctx context.Context, bundle *common.Bundle) error
```

Comment:
I think we must avoid to use common protos on plugins,
what do you think about taking the same approach we took on some new plugin types,
where we create structs that we use as parameters? for example in credential composer or keymanager

Reply:
My vote would be to keep this as-is since it aligns with that the datastore returns. Any other types right now would be unnecessary friction/conversion for no benefit IMO.